lib/fog/dynect/models/dns/record.rb

```diff
@@ -25,7 +25,9 @@ def save(replace=false)
           }
           options.delete_if {|key, value| value.nil?}
 
-          if replace
+          refresh_id if replace && identity.nil?
+          if replace && !identity.nil?
```

Comment:
I think you are right then. The whole problem is having a boolen param that change the behaviour of the method. It makes it hard to follow and understand. 

Maybe in a future refactoring and major bump we can change that to be two different methods that completely express its behaviour e.g (`save`, `overwrite`).